### PR TITLE
Italian translation

### PR DESCRIPTION
**Pull request by Feyn-Man**
italian translation for #1567

### CODE REVIEW

**Comment by mgrojo** [comment]
You are on the correct path. You can continue the translation and commit the updated file. When you've finished, let us know.

**Comment by justinclift** [comment]
@Feyn-Man How's this going?  We'll probably create a next beta release (beta2) tomorrow or so.

Are the files you've already done, ok to be merged into that? :smile:

**Comment by Feyn-Man** [comment]
Hi @justinclift, sorry for this enormous period of absence! I had a lot of work to do and this slipped out of my mind...I will restart to push soon...files already pushed are ready :)

**Comment by justinclift** [comment]
Fantastic.  I'll get this merged today or tomorrow, in time for our next beta release, if no-one else gets to it before me. :smile:

There are a few minor side pieces we need to do, for enabling this as well:

* [ ] Add the Italian flag (from [here](https://github.com/joielechong/iso-country-flags-svg-collection)), so it's shown in the drop down language selection list.
* [ ] Add an entry to the [CMake build file](https://github.com/sqlitebrowser/sqlitebrowser/blob/6425fb177eba293ae000415392d088af96e27c91/CMakeLists.txt#L234-L250) as well
  * We support both CMake and qmake, so both get an entry for this. :smile:
* [ ] Add an entry to the [translations list file here](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/translations/translations.qrc) too.
* [ ] Double check the flag shows up correctly in both the windows and macOS installers.
  * We've had cases before where the initial attempt didn't work 100%, so needed adjustment.  Thus the double checking just to be safe.

**Comment by Feyn-Man** [comment]
Ok. Only 2 questions: 

1. How can I check that the flag shows correctly in the installers?
2. So I do all this work on the master branch and then I commit to the `v3.11.x` branch? Is this correct?

Forgive me for these maybe stupid questions, but I'm new to this type of things.

**Comment by justinclift** [comment]
> How can I check that the flag shows correctly in the installers?

Oops.  Incorrect wording on my part there.  What I meant was we need to check if the installed packages (on windows and macOS) show the Italian flag correctly in the DB4S language selection drop down, in our Preferences dialog.  eg generate a new nightly build with the above pieces done, install it, start DB4S, check if the Italian flag is shown in our language selector.  For both windows and macOS.

If you have windows and macOS available for testing, then yep it's something you can do.  If not, no worries as I can get it done. :smile:

> So I do all this work on the master branch and then I commit to the v3.11.x branch? Is this correct?

Almost.  You do the work on the master branch, and commit to the master branch.  The same commit can then be applied to other branches too (that's called "cherry-picking" a commit).  So we apply it to the `v3.11.x` branch as well, which our 3.11.0 release is being created from. :smile:

> Flags are taking so much time in generating, is this ok? It seems strange to me....

Ahh.  I don't think you need to run anything.  From memory of the last time I did it, I *think* I used the Gimp (like Photoshop) to import an svg image of a flag from that repo, then exported it as a png and added that to our repo.  May have needed to resize it to, not sure.  The flag needs to be the same size as the other ones in our repo.

The flag filename also gets added here too: https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/translations/flags/flags.qrc

From memory, that last bit - adding the flag to flags.qrc - was the tricky one.  Most of the time it's ok, but **sometimes** it takes a bit of mucking around.

**Comment by Feyn-Man** [comment]
Ok. I have done the changes. I've committed and pushed on the master of my fork (maybe the push wasn't necessary, I've done it as an automatism). Now, what should I do? Create a pull and then you cherry-pick?

For the flags, yesterday eve I didn't see that there was an svg folder, too late and too tired :sweat_smile: I was generating the pngs of the flags of all the world in every format from that repo ahah

**Comment by justinclift** [comment]
Ahhh.  The changes should probably go in the `italian-translation` branch of your fork.  That's what this PR is using, so that's what'll get merged into our `master`.

Alternatively, if you want them in a different PR to this one... then um... create a new PR but choose whichever branch you chucked them in.  Sounds like it'll be the `master` branch of your fork.

If this gets a bit too confusing, just let me know.  It's late in the evening here and I've not had a chance to do anything at all with DB4S today (fixing infrastructure stuff instead).  So, I'm going to get some sleep soonish.  I can take a look at the state of your branches tomorrow and pick out the bits needed then if that's better.  The building of the beta2 release can come after that. :smile:

**Comment by Feyn-Man** [comment]
Ok, sorry Justin, my mistake, I misunderstood your instructions. I will remake the same changes on the italian-branch and pull here only the italian-branch whith the changes needed when I return home. Anyway thank you for your patience, this is being instructive for me.

**Comment by Feyn-Man** [comment]
Ok, here's the commit: I added the flag and made the changes needed to the files. Sorry again for the late, tell me if all it's ok. Should I create a new branch to continue translating after you have merged?

**Comment by justinclift** [comment]
> Sorry again for the late, tell me if all it's ok.

Will do.  I need to go out for a bit now, but will return to this later on today and check. :smile:

> Should I create a new branch to continue translating after you have merged?

Yep, once this is merged you can delete the branch you were doing it in as it's no longer needed.  The changes will then be in our `master` branch, so you'll want to `git pull` in your local repo to update your repo with all of the changes that have happened in ours (your commits + those of other people).

To continue translating afterwards, just make a new branch (based on the updated repo mentioned above).  Put your new commits in the new branch, and make a PR with it.

Hopefully I'm explaining that ok.  I'm a bit mentally brain drained at the moment. :innocent:

File: src/translations/translations.qrc [hunk 1]
```diff
@@ -14,5 +14,6 @@
         <file alias="sqlb_ko_KR">sqlb_ko_KR.qm</file>
         <file alias="sqlb_tr">sqlb_tr.qm</file>
         <file alias="sqlb_uk_UA">sqlb_uk_UA.qm</file>
+	<file alias="sqlb_it">sqlb_it.qm</file>
```
Comment (justinclift) [inline src/translations/translations.qrc]: Seems to be a mismatch with spaces vs tabs here.

In this instance don't worry about fixing it.  I'll merge these commits now and manually tweak it right after as I'm just about to build a few test nightly builds anyway. :smile:

**Comment by justinclift** [comment]
Just updated that tab-vs-spaces thing, and manually started out Windows 64-bit nightly build script.  That generally takes about 1/2 hour to finish + upload the build, assuming nothing goes weird. :smile:

We can  use that nightly build to check if the flag is being shown correctly.  If it is, then I'll launch the Windows 32-bit nightly build script to make the 32-bit version, then the macOS one too.  We'd better check the flag in both of those as well.  (I have Win+macOS stuff here, so can do it)

**Comment by justinclift** [comment]
Oh.  The flag isn't showing up in the drop down list:

italian-missing_flag_win64

It'll investigate in a bit.  Updating our infrastructure servers at the moment.

**Comment by justinclift** [comment]
Ahhh, think I found it.  Just a missing entry in the [flags.qrc](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/translations/flags/flags.qrc) file.

I'll update that and rebuild the nightly.  Hopefully that's all it is. :smile:

**Comment by justinclift** [comment]
Yep, that was it:

italian-flag_working_win64

I'll try the Win32 and macOS builds next. :smile:

**Comment by justinclift** [comment]
Found something that might be a typo, but not sure.

In the screenshot below, it looks like there are some spaces missing on the line with the SQLCipher & SQLite version numbers?

italian-missing_space_between_strings

Here's the same dialog, but with the program switched to English:

english-showing_space_between_strings

It's not at all urgent, but at some point we should investigate and figure out how to fix it. :smile:

File: src/translations/translations.qrc (hunk 1 above)
Comment (Feyn-Man) [reply src/translations/translations.qrc]: Yes, I used tab, probably there were spaces but the editor doesn't show the difference.

**Comment by Feyn-Man** [comment]
> Ahhh, think I found it. Just a missing entry in the [flags.qrc](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/translations/flags/flags.qrc) file.
>

Ok, just add this passage in the instructions for adding the flag :smile: 




> Found something that might be a typo, but not sure.
> 
> In the screenshot below, it looks like there are some spaces missing on the line with the SQLCipher & SQLite version numbers?
> 

Ok, I'll check soon if it's a typo.

**Comment by Feyn-Man** [comment]
> Ok, I'll check soon if it's a typo.

It's a typo :smile:

**Comment by justinclift** [comment]
Win32 build looks good.  Flag is shown correctly.  macOS build to do next. :smile:

**Comment by justinclift** [comment]
macOS is displaying the Italian flag properly in the language selection piece as well.

Looks like we're ok. :smile:

Just need to get the remaining strings translated to Italian, when you have the time + energy (at some future point).

Thanks heaps for this @Feyn-Man. :smile:

**Comment by mgrojo** [comment]
Ok, it's working now. I don't know if has been something that I've done, or it was working also before, but I didn't notice any of the translated parts.

**Comment by justinclift** [comment]
@Feyn-Man Cool. :smile:

@mgrojo This is only a partial translation (so far), so only some of the pieces are in place.

When I was testing the Win64 build, much of the UI was still shown in English. The dialog box was all translated though, so I think it's just a case of the main UI strings not yet being done. :wink:

**Pull request merged**